### PR TITLE
Communikate CK20Page refactor and clean

### PR DESCRIPTION
**Pull request by zelmon64**
Refactor and possible bug fixes in the CK20Page class(es) as outlined in #375.

This mainly targets `CKPageFileChanged` but also improves `dec2hex` and the background colour setters.

### CODE REVIEW

File: src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs [hunk 1]
```diff
@@ -1130,7 +622,7 @@ public string CKText_10
         public string CKBaCo_10
         {
             get { return (string)GetValue(CKBaCo_10Property); }
-            set { SetValue(CKBaCo_10Property, dec2hex(value)); }
+            set { SetValue(CKBaCo_10Property, value); }
         }
 
         public static readonly DependencyProperty CKImSo_10Property =
```
Comment (kmcnaught) [inline src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs]: For each of the CKImSo_xy setters (a few lines below, I can't comment there) could you please move the logic into one place outside of the setter, just like you did for rgb2hex. This makes it easier to modify if the logic changes later.

**Comment by kmcnaught** [comment]
Hi @zelmon64 , could you please also remove the unused CKKey files from the project in this PR?

File: src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs [hunk 2]
```diff
@@ -68,46 +68,48 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
...
-                    blankbutton.label = defaulttext;
-                    //blankbutton.load_board.path = defaultpath;
-                    blankbutton.load_board = defaultboard;
+                    blankbutton.background_color = defaultColour;
+                    blankbutton.image_id = defaultImage;
+                    blankbutton.label = defaultText;
+                    blankbutton.load_board = defaultBoard;
 
                     for (int r = 1; r < 5; ++r)
+                    {
```
[the remark this answers is not in the document]
Comment (zelmon64) [reply src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs]: I've added some comments but also rearranged it to a simpler method. If the comments aren't enough, please say so and I'll add more.

File: src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs [hunk 3]
```diff
@@ -68,46 +68,48 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
...
                         for (int c = 0; c < 5; ++c)
                         {
-                            if (b < CKPageOBF.buttons.Count)
+                            if (ButtonNo < CKPageOBF.buttons.Count)
                             {
-                                if (CKPageOBF.buttons.ElementAt(b).id != c.ToString() + r.ToString())
+                                if (CKPageOBF.buttons.ElementAt(ButtonNo).id != c.ToString() + r.ToString())
                                 {
                                     if (BlankButtonCount < 2)
```
[the remark this answers is not in the document]
Comment (zelmon64) [reply src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs]: `if (BlankButtonCount == 1 )` does make more sense so I've changed it to that.

File: src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs [hunk 4]
```diff
@@ -124,12 +126,13 @@ private static void CKPageFileChanged(DependencyObject dependencyObject, Depende
                                         Log.DebugFormat("Blank button {3} added at column {0} row {1} with background colour {2}.", c, r, blankbutton.background_color, BlankButtonCount);
                                     }
                                     blankbutton.id = c.ToString() + r.ToString();
```
[the remark this answers is not in the document]
Comment (zelmon64) [reply src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs]: The inserted buttons are now individually instantiated and the properties are immediately set. The `Buttons` class now also contains defined defaults.

File: src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs (hunk 1 above)
Comment (zelmon64) [reply src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs]: Thanks for reminding me about this. The logic is now done on lines [168-170](https://github.com/zelmon64/OptiKey/blob/29b1d2ace719f6e1f8a2f0b50537a164227aab2b/src/JuliusSweetland.OptiKey/UI/Controls/CK20Page.xaml.cs#L168-L170).

**Comment by zelmon64** [comment]
@kmcnaught thanks for your comments/suggestions. I agree with all of them and have modified the code accordingly. I also simplified where the blank keys are inserted and added default values to the `Buttons` class.

I would love your thoughts on these changes.

**Comment by kmcnaught** [comment]
@zelmon64 Great, thanks, I'm happy with everything now except for the spelling of the new variable `CurentButtonIsBlank` :)

**Pull request merged**